### PR TITLE
Restrict without description

### PR DESCRIPTION
**Pull request by Checho3388**
# Description
Added support for restricting changes in schema.

If argument `--restrictions` is provided after calculating changes, it will evaluate if any of the changes is forbidden by the rule names provided as arguments. If there's at least one forbidden change, exit code is `1`.

By now, two built-in restrictions are created:
- Restrict adding types without description
- Restrict removing type description

Restricted changes are displayed like this:
```
⛔ Type `NewTypeWithoutDesc` was added without a description
```

### CODE REVIEW

**Comment by Ambro17** [comment]
Hi Eze, thanks for your PR!
I think codecov is complaining because of this line 
https://github.com/Ambro17/graphql-schema-diff/blob/restrict_without_description/schemadiff/restrictions/__init__.py#L19

Try decorating the `@classmethod` *also* with `@abstractmethod` and remove that line as it will never be reached (And with the docstring is enough to get valid syntax)

**Comment by Ambro17** [comment]
Codecov complains because some lines are missing tests.
Here you can see the details
https://codecov.io/gh/Ambro17/graphql-schema-diff/compare/c254248aa77a41b1125d51a69d518ba5e3df1c57...fb8a14c776f39a76ec2031ec5750bdce6a13d0cb/diff

**Comment by Checho3388** [comment]
Everything is green now!

File: tests/test_restrictions.py [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    a = schema("""
+    type MyType{
+        a: Int
+    }
+    """)
+    b = schema("""
+    type MyType{
+        a: Int
+    }
+    \"\"\"This has desc\"\"\"
```
[the remark this answers is not in the document]
Comment (Checho3388) [reply tests/test_restrictions.py:42]: :+1: Done

File: tests/test_restrictions.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        a: Int
+    }
+    """)
+    b = schema("""
+    type MyType{
+        a: Int
+    }
+    """)
+    c = schema("""
+    \"\"\"\"\"\"
```
[the remark this answers is not in the document]
Comment (Checho3388) [reply tests/test_restrictions.py:82]: :+1: Done

File: schemadiff/formatting.py [hunk 3]
```diff
@@ -21,7 +21,7 @@ def format_change_by_criticality(change: Change) -> str:
         CriticalityLevel.NonBreaking: os.getenv('SD_SAFE_CHANGE_ICON', '✔️'),
     }
     icon = icon_by_criticality[change.criticality.level]
-    return f"{icon} {change.message}"
+    return f"{icon} {change.message}{' ⬅️ ⛔ Restricted' if change.restricted else ''}"
```
[the remark this answers is not in the document]
Comment (Checho3388) [reply schemadiff/formatting.py]: Ok. Added a change to include a more descriptive message!

File: schemadiff/restricted_changes.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+from typing import List
+
+from schemadiff.restrictions import Restriction
```
[the remark this answers is not in the document]
Comment (Checho3388) [reply schemadiff/restricted_changes.py]: Ok. I guess all renaming is done!

File: tests/data/simple_schema_rules_validation_new.gql [hunk 5]
```diff
@@ -18,4 +18,10 @@ type NewTypeWithoutDesc {
 
 enum  NewEnumWithoutDesc {
     VALUE
+}
+
+enum Enum {
+    VALUE_1
+    VALUE_2
+    VALUE_3
 }
```
Comment (Ambro17) [inline tests/data/simple_schema_rules_validation_new.gql:27]: Missing newline here

**Pull request merged**